modules/zarr/src/lib/utils.ts

```diff
@@ -0,0 +1,147 @@
+import { openGroup } from 'zarr';
+import type { ZarrArray } from 'zarr';
+import type { Store } from 'zarr/types/storage/types';
+
+import { FetchFileStore } from './storage';
+import type { PixelSource, RootAttrs, Labels } from '../types';
+
+export function normalizeStore(source: string | Store): Store {
+  if (typeof source === 'string') {
+    return new FetchFileStore(source);
+  }
+  return source;
+}
+
+export async function loadMultiscales(store: Store, path = '') {
+  const grp = await openGroup(store, path);
+  const rootAttrs = await grp.attrs.asObject() as RootAttrs;
+
+  if (!Array.isArray(rootAttrs.multiscales)) {
```

Comment:
checks that multiscales extension is implemented in rootAttrs (generic across Zarr datasets)

Reply:
A Zarr store is organized into a hierarchy of nodes. Each node can either be a `group` or an `array`, and only a `group` may contain other nodes. The multiscales extension is _not_ OME-Zarr -specific (although it was spearheaded by OME folks working on zarr). 

The mutiscales extension adds a `"multiscales"` key-value to `group` metadata that ultimately provides the relative paths to other array sub-resolutions saved in the store. OME-Zarr chooses to also store additional metadata in the same group but under a different key (`omero`). This function reads the multiscales metadata (to initialize the pixel-sources), but simply returns the root metadata back to the end-user to choose what to do with it.

Reply:
Sorry for the late addition, but there has been off and on discussion of moving the multiscales to a "super-interface" of OME-Zarr, i.e. a more generalized pyramidal image spec.

cf. https://github.com/pydata/xarray/issues/4118#issuecomment-873179375